### PR TITLE
refactor: disallow DELEGATECALL operation in LSP9Vault

### PR DESCRIPTION
**Pull request by YamenMerhi**
## What does this PR introduce?
- Overriding execute function in LSP9Vault and disallowing the use of delegatecall, since it's a dangerous operation and could lead to the destruction of the vault.

The use of delegatecall in LSP9 is totally optional. 
Referring to the [LSP9 specs](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-9-Vault.md), it's mention that an LSP9Vault must include the following operation: CALL, CREATE, CREATE2, STATICCALL and CAN include the operation DELEGATECALL.

### CODE REVIEW

**Review by b00ste — APPROVED** [review]

File: contracts/LSP9Vault/LSP9VaultCore.sol [hunk 1]
```diff
@@ -3,6 +3,7 @@ pragma solidity ^0.8.0;
 
 // interfaces
 import {IERC725Y} from "@erc725/smart-contracts/contracts/interfaces/IERC725Y.sol";
```
Comment (CJ42) [inline contracts/LSP9Vault/LSP9VaultCore.sol:5]: You can remove these two lines from the imports (` import {IERC725X} ...` and `import {IERC725Y} ...`). They are not used in the contract anywhere (neither in the inheritance, nor inside the contract code.

File: contracts/LSP9Vault/LSP9VaultCore.sol [hunk 2]
```diff
@@ -21,6 +22,8 @@ import {ClaimOwnership} from "../Custom/ClaimOwnership.sol";
 
 // constants
 import {_INTERFACEID_CLAIM_OWNERSHIP} from "../Custom/IClaimOwnership.sol";
+
+import "@erc725/smart-contracts/contracts/constants.sol";
```
Comment (CJ42) [inline contracts/LSP9Vault/LSP9VaultCore.sol:26]: Why not adding the exact constants imported here, so they can be easily found?

```solidity
import {
    OPERATION_CALL, 
    OPERATION_STATICCALL, 
    OPERATION_CREATE, 
    OPERATION_CREATE2
} from "@erc725/smart-contracts/contracts/constants.sol";
```

File: contracts/LSP9Vault/LSP9VaultCore.sol [hunk 3]
```diff
@@ -74,61 +77,40 @@ contract LSP9VaultCore is ERC725XCore, ERC725YCore, ClaimOwnership, ILSP1Univers
...
     }
 
-    // ERC165
+    // ERC725
 
     /**
-     * @dev See {IERC165-supportsInterface}.
+     * @inheritdoc IERC725X
+     * @param operation The operation to execute: CALL = 0 CREATE = 1 CREATE2 = 2 STATICCALL = 3
+     * @dev Executes any other smart contract.
```
Comment (CJ42) [inline contracts/LSP9Vault/LSP9VaultCore.sol:85]: would it be worth mentioning in the Natspec docs as well that DELEGATECALL is disallowed?

File: contracts/LSP9Vault/LSP9VaultCore.sol [hunk 4]
```diff
@@ -208,7 +243,7 @@ contract LSP9VaultCore is ERC725XCore, ERC725YCore, ClaimOwnership, ILSP1Univers
     }
 
     /**
-     * @dev Calls the universalReceiver function of the sender if supports LSP1 InterfaceId
+     * @dev Calls the universalReceiver function of the vault sender if supports LSP1 InterfaceId
```
Comment (CJ42) [inline contracts/LSP9Vault/LSP9VaultCore.sol:246]: nice extra specs 👍

File: contracts/LSP9Vault/LSP9VaultCore.sol (hunk 2 above)
Comment (CJ42) [inline contracts/LSP9Vault/LSP9VaultCore.sol:26]: Why not adding the exact constants imported here, so they can be easily found?

```solidity
import {
    OPERATION_CALL, 
    OPERATION_STATICCALL, 
    OPERATION_CREATE, 
    OPERATION_CREATE2
} from "@erc725/smart-contracts/contracts/constants.sol";
```

**Review by CJ42 — CHANGES_REQUESTED** [review]

File: contracts/LSP9Vault/LSP9VaultCore.sol (hunk 1 above)
Comment (YamenMerhi) [reply contracts/LSP9Vault/LSP9VaultCore.sol:5]: They are used within the natspec

File: contracts/LSP9Vault/LSP9VaultCore.sol (hunk 2 above)
Comment (YamenMerhi) [reply contracts/LSP9Vault/LSP9VaultCore.sol:26]: Good pt

File: contracts/LSP9Vault/LSP9VaultCore.sol (hunk 3 above)
Comment (YamenMerhi) [reply contracts/LSP9Vault/LSP9VaultCore.sol:85]: I overrided the @param tag to point out that these are the operation allowed. I guess this is a good base

**Pull request merged**